### PR TITLE
Stop using sprintf() in SharedCacheAPITest.cpp

### PR DESCRIPTION
**Pull request by knn-k**
This commit rewrites getCacheDir() in SharedCacheAPITest.cpp for
eliminating sprintf().

Signed-off-by: KONNO Kazuhiro <konno@jp.ibm.com>

### CODE REVIEW

**Comment by knn-k** [comment]
This is a part of fixes for Issue #16517.
There are many other files that use `sprintf()`, but it seems SharedCacheAPITest.cpp is the only file in the OpenJ9 project that blocks the build with Xcode 14 at the moment.

**Comment by knn-k** [comment]
Jenkins test sanity alinux64,amac jdk17 depends eclipse/omr#6862

File: runtime/tests/shared/SharedCacheAPITest.cpp [hunk 1]
```diff
@@ -59,7 +59,7 @@ getCacheDir(J9JavaVM *vm, char *cacheDir)
 		j9tty_printf(PORTLIB, "Cannot get a directory\n");
 	}
 
-	sprintf(cacheDir, "%s%s", cacheDirNoTestBasedir, TEST_BASEDIR);
+	snprintf(cacheDir, bufLen, "%s%s", cacheDirNoTestBasedir, TEST_BASEDIR);
```
Comment (keithc-ca) [inline runtime/tests/shared/SharedCacheAPITest.cpp:62]: A more robust solution would pass a smaller length to `j9shmem_getDir()` so there would be no possibility of overflow adding `TEST_BASEDIR` (and eliminate the need for the second buffer `cacheDirNoTestBasedir`):
```cpp
void
getCacheDir(J9JavaVM *vm, char *cacheDir, size_t bufLen)
{
	PORT_ACCESS_FROM_JAVAVM(vm);

	cacheDir[0] = '\0';

	if (bufLen < LITERAL_STRLEN(TEST_BASEDIR)) {
		j9tty_printf(PORTLIB, "Buffer too small to get directory\n");
	} else {
		U_32 flags = J9SHMEM_GETDIR_APPEND_BASEDIR
#if defined(OPENJ9_BUILD) && !defined(J9ZOS390)
				| J9SHMEM_GETDIR_USE_USERHOME
#endif /* defined(OPENJ9_BUILD) && !defined(J9ZOS390) */
				;
		IDATA rc = j9shmem_getDir(NULL, flags, cacheDir, bufLen - LITERAL_STRLEN(TEST_BASEDIR));
		if (rc < 0) {
			j9tty_printf(PORTLIB, "Cannot get a directory\n");
		} else {
			strcat(cacheDir, TEST_BASEDIR);
		}
	}
}
```

**Comment by knn-k** [comment]
Jenkins test sanity xlinux,amac jdk17

File: runtime/tests/shared/SharedCacheAPITest.cpp (hunk 1 above)
Comment (knn-k) [reply runtime/tests/shared/SharedCacheAPITest.cpp]: Thank you.
I updated `getCacheDir()` as suggested, with one minor change:
`if (bufLen < LITERAL_STRLEN(TEST_BASEDIR))` -> `if (bufLen <= LITERAL_STRLEN(TEST_BASEDIR))`

**Comment by knn-k** [comment]
It seems build machines are not available today.

**Comment by keithc-ca** [comment]
Jenkins test sanity zlinux jdk17

**Comment by knn-k** [comment]
Jenkins test sanity xlinux,amac jdk17

**Review by keithc-ca — APPROVED** [review]

**Pull request merged**